app/src/main/java/com/tari/android/wallet/infrastructure/backup/WalletBackup.kt

```diff
@@ -0,0 +1,113 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.tari.android.wallet.infrastructure.backup
+
+import com.tari.android.wallet.infrastructure.backup.compress.CompressionMethod
+import com.tari.android.wallet.infrastructure.security.encryption.SymmetricEncryptionAlgorithm
+import com.tari.android.wallet.util.Constants
+import net.lingala.zip4j.ZipFile
+import net.lingala.zip4j.model.ZipParameters
+import net.lingala.zip4j.model.enums.AesKeyStrength
+import net.lingala.zip4j.model.enums.EncryptionMethod
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+
+
+interface WalletBackup {
+    fun run(key: CharArray): File
+
+    companion object {
+        fun defaultStrategy(workingDir: File): WalletBackup = zipOnly(workingDir)
+
+        fun passwordProtectedZip(workingDir: File): WalletBackup =
+            PasswordProtectedZipBackup(workingDir)
+
+        fun aesEncryptedZip(workingDir: File): WalletBackup = CompositeWalletBackup(
+            workingDir,
+            SymmetricEncryptionAlgorithm.aes(),
+            CompressionMethod.zip()
+        )
+
+        fun zipOnly(workingDir: File): WalletBackup =
+            CompressionOnlyBackup(workingDir, CompressionMethod.zip())
+
+    }
+}
+
+val BACKUP_FILES = arrayOf(Constants.Wallet.walletDBFullFileName)
+
+private class PasswordProtectedZipBackup(private val workingDir: File) : WalletBackup {
+    override fun run(key: CharArray): File {
+        val zipFile = ZipFile("${workingDir.absolutePath}/backup_${System.currentTimeMillis()}.zip")
+        val parameters = ZipParameters()
+        if (key.isNotEmpty()) {
+            parameters.isEncryptFiles = true
+            parameters.encryptionMethod = EncryptionMethod.AES
+            parameters.aesKeyStrength = AesKeyStrength.KEY_STRENGTH_256
+            zipFile.setPassword(key)
+        }
+        zipFile.addFiles(BACKUP_FILES.map { File(workingDir, it) }.toMutableList(), parameters)
+        return zipFile.file
+    }
+}
+
+private class CompositeWalletBackup(
+    private val workingDir: File,
+    private val encryptionAlgorithm: SymmetricEncryptionAlgorithm,
+    private val compressionMethod: CompressionMethod
+) : WalletBackup {
+
+    override fun run(key: CharArray): File {
+        val backupTime = System.currentTimeMillis()
+        val archive = compressionMethod.compress(
+            "${workingDir.absolutePath}/temp_$backupTime.${compressionMethod.extension}",
+            BACKUP_FILES.map { File(workingDir, it) })
+        val encryptedArchiveFile = File(workingDir, "backup_$backupTime")
```

Comment:
Let's follow the same naming as iOS: `Tari-Aurora-Backup_yyyy-MM-dd_HH-mm-ss`

Reply:
this is a temp file that will be removed a line after, the file on the drive has the aforementioned naming: https://github.com/tari-project/wallet-android/pull/435/files#diff-ee7d944a78387dadce87252cc41def1cR121